CRM/Contribute/Form/ContributionBase.php

```diff
@@ -293,7 +293,7 @@ public function preProcess() {
       $isPayLater = CRM_Utils_Array::value('is_pay_later', $this->_values);
 
       if ($isMonetary &&
-        (!$isPayLater || !empty($this->_values['payment_processor']))
+        (!$isPayLater && !empty($this->_values['payment_processor']))
```

Comment:
The issue with 'Execute real time monetary transactions' = FALSE and pay_later= TRUE setting where no membership and related contribution was created earlier, was fixed in https://github.com/civicrm/civicrm-core/pull/8575/files#diff-99d052aa6a966c5a2cc89d05f057d22dR234.